src/ar_gateway_h.erl

```diff
@@ -0,0 +1,140 @@
+-module(ar_gateway_h).
+-behaviour(cowboy_handler).
+-export([init/2]).
+
+-include("ar.hrl").
+-define(BH_SEARCH_TIMEOUT, 5000).
```

Comment:
Only waiting 5s sounds too short for me. I have successfully waited more than 5s for a webpage to load many times.

Reply:
So let's go for `25000` 👍